pmd-core/src/main/java/net/sourceforge/pmd/PMD.java

```diff
@@ -350,31 +349,6 @@ public static RuleContext newRuleContext(String sourceCodeFilename, File sourceC
     }
 
     /**
-     * An entry point that would typically be used by IDEs intent on providing
-     * ongoing feedback and the ability to terminate it at will.
-     *
-     * @param configuration
-     *            the PMD configuration to use
-     * @param ruleSetFactory
-     *            ruleset factory
-     * @param files
-     *            the files to analyze
-     * @param ctx
-     *            the rule context to use for the execution
-     * @param monitor
-     *            PMD informs about the progress through this progress monitor.
-     *            It provides also the ability to terminate/cancel the
-     *            execution.
-     */
-    public static void processFiles(PMDConfiguration configuration, RuleSetFactory ruleSetFactory,
```

Comment:
We should remove also the interface ProgressMonitor, which is defined above (lines 335-350).